### PR TITLE
[`flake8-pyi`] Skip type annotations in `string-or-bytes-too-long` (`PYI053`)

### LINKED ISSUE #12995: `string-or-bytes-too-long (PYI053)` flags and removes long strings in `Literal`, producing invalid type
In the example reproduction below, 

* List of keywords you searched for before creating this issue. Write them down here so that others can find this issue more easily and help provide feedback.
  string-or-bytes-too-long PYI053
  
* A minimal code snippet that reproduces the bug.
```py
def foo(bar: typing.Literal["a", "bbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbbb"]):...
```
becomes
```py
import typing

def foo(bar: typing.Literal["a", ...]):...
```

Not only is this an invalid type form (https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportInvalidTypeForm) but even if autofixed to `Literal["a"]` that would fundamentally change the parameter's type.

I assume this is an oversight as it's meant to replace long default values ~~and is not flagged by flake8-pyi~~ .

* The command you invoked (e.g., `ruff /path/to/file.py --fix`), ideally including the `--isolated` flag.
ruff check --isolated --select=PYI053

* The current Ruff settings (any relevant sections from your `pyproject.toml`).
not relevant

* The current Ruff version (`ruff --version`).
ruff 0.6.1

### PR DESCRIPTION
**Pull request by dylwil3**
PYI053 is meant to protect against default arguments being too lengthy, not, e.g., type annotations with long literal strings.

Closes #12995

### CODE REVIEW

File: crates/ruff_linter/src/rules/flake8_pyi/rules/string_or_bytes_too_long.rs [hunk 1]
```diff
@@ -59,6 +59,10 @@ pub(crate) fn string_or_bytes_too_long(checker: &mut Checker, string: StringLike
         return;
     }
 
+    if semantic.in_annotation() {
```
Comment (AlexWaygood) [inline crates/ruff_linter/src/rules/flake8_pyi/rules/string_or_bytes_too_long.rs:62]: I wondered if it might be better to be more targeted in our exclusions here, since there are relatively few valid annotations in stub files that would involve long string literals -- basically just `Literal` and `Annotated`, I think? We can detect if we're in a `typing.Literal` slice like this:

```suggestion
    if semantic.in_typing_literal() {
```

But we don't currently track in the semantic model whether we're in an `Annotated` slice in the same way, and I'm not sure if it's worth it to add that tracking just for this, as I'm not sure it would practically get us much.

TL;DR -- this seems good; thanks!

**Review by AlexWaygood — APPROVED** [review]
Thanks!
